pkg/apis/pipeline/v1alpha2/pipeline_types.go

```diff
@@ -227,3 +232,29 @@ type PipelineList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Pipeline `json:"items"`
 }
+
+// PipelineTaskRunOn represents runOn of a pipelineTask which is a map of task and a list of states, e.g.
+// runOn:
+// 	- task: task1
+//    states: ["success", "failure"]
+//  - task: task2
+//    states: ["success"]
+type PipelineTaskRunOn struct {
+	// Task is the name of the PipelineTask as specified in a pipeline
+	Task string `json:"task"`
+	// States is a list of states of a PipelineTask which is specified in the Task above
+	States []PipelineTaskState `json:"states"`
+}
+
+// PipelineTaskState represents the state of a parent PipelineTask that it has to be in for it to run
+type PipelineTaskState = string
+
+const (
+	// PipelineTaskStateSuccess indicates that a PipelineTask has finished execution successfully
+	PipelineTaskStateSuccess PipelineTaskState = "success"
+	// PipelineTaskStateFailure indicates that a PipelineTask has finished execution and has failed
+	PipelineTaskStateFailure PipelineTaskState = "failure"
+)
```

Comment:
Adding two possible states to begin with: (1) success (2) failure, remaining two states (skip and always) will go here

Reply:
I think other states we'd need are 1. skip for condition failure 2. skip due to a previously failing taskrun

Reply:
humm, `skip` for sure, and may be `always`?

`skip` would run when parent tasks skipping due to condition failure.

Not sure what (2) signify? By default, `runOn` task is skipped if the states conflicts and that task is marked as conflicted (`RunOnStateConflicted`), any task dependent on task with `RunOnStateConflicted` is also skipped and marked `RunOnStateConflicted` instead of failing the whole pipeline, please check pipeline.md for more details.